### ISSUE TITLE
Validate UpdatePolicy for AutoScalingRollingUpdate 

### ISSUE THREAD

**Issue by adamchainz** [aws-cloudformation/cfn-lint#362]
*cfn-lint version: (`cfn-lint --version`)* 0.7.2

*Description of issue.*

Using a template like:

```yaml
Resources:
  AutoScalingGroup:
    Type: AWS::AutoScaling::AutoScalingGroup
    Properties:
      # ...
      MinSize: "1"
      MaxSize: "1"
      # ...
    UpdatePolicy:
      AutoScalingRollingUpdate:
        MinInstancesInService: 1
```

will fail at deployment time with an error from CloudFormation like:

> AWS::AutoScaling::AutoScalingGroup AutoScalingGroup UPDATE_FAILED: MinInstancesInService must be less than the autoscaling group's MaxSize

It would be nice to have protection against this. I guess there are other constraints, some are [in the docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html#cfn-attributes-updatepolicy-rollingupdate). I submitted a [docs PR for this one](https://github.com/awsdocs/aws-cloudformation-user-guide/pull/164).

**Comment by cmmeyer** [comment]
Agreed.  This would be a good business logic addition to compare values.  I'll label this as an enhancement.

**Comment by kddejong** [comment]
Started with rule E3016 to check the basic configuration of the UpdatePolicy (attributes, values are of the correct type), etc.

Moving down the chain to this one next.

**Comment by kddejong** [comment]
Sorry @kalpik I lost track of this.  I have created pull request #468 totally rewritten to handle conditions on each level.  Used your example for adding to tests.  Thanks for reporting this.

**Comment by digash** [comment]
I am getting error E3016 UpdatePolicy doesn't support type AutoScalingRollingUpdate from this AWS example:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/quickref-autoscaling.html#scenario-as-updatepolicy

**Comment by kddejong** [comment]
Closing this issue as it is covered in v1.   Any improvements we can do we will do as requested.
